### PR TITLE
Add withKeyMultiValues to MutableMultimap and subtypes. Closes #1307.

### LINKED ISSUE #1307: Add withKeyMultiValues to MutableMultimap and subtypes
In addition to `withKeyValue`, which I describe here (https://github.com/eclipse/eclipse-collections/issues/1303), it would be useful to have a specialized `withKeyMultiValues` method which will take a single key and multiple values to add to the `Multimap`.

```
MutableListMultimap<String, String> multimap = Multimaps.mutable.list.empty()
        .withKeyMultiValues("A", Lists.mutable.with("Apple", "Apricot"))
        .withKeyMultiValues("B", Lists.mutable.with("Banana", "Blueberry"))
        .withKeyMultiValues("C", Lists.mutable.with("Cantaloupe", "Clementine"));
```

This could also be implemented via varargs.

```
MutableListMultimap<String, String> multimap = Multimaps.mutable.list.empty()
        .withKeyMultiValues("A", "Apple", "Apricot")
        .withKeyMultiValues("B", "Banana", "Blueberry")
        .withKeyMultiValues("C", "Cantaloupe", "Clementine");
```

### PR DESCRIPTION
**Pull request by goldbal330**
Signed-off-by: Alexander Goldberg <alexander.goldberg@bnymellon.com>

### CODE REVIEW

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java [hunk 1]
```diff
@@ -35,6 +35,24 @@ default MutableBagIterableMultimap<K, V> withKeyValue(K key, V value)
...
+    default MutableBagIterableMultimap<K, V> withKeyMultiValues(K key, V... values)
+    {
+        if (values != null && values.length > 0)
+        {
+            for (V value : values)
+            {
+                this.put(key, value);
+            }
+        }
+        return this;
```
Comment (motlin) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java:53]: In overrides, you could have a similar implementation just like the one I suggested above. Another alternative is to call super and cast.

```java
return (MutableBagIterableMultimap<K, V>) MutableMultimap.super.withKeyMultiValues(key, values);
```

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java [hunk 2]
```diff
@@ -35,6 +35,24 @@ default MutableBagIterableMultimap<K, V> withKeyValue(K key, V value)
         return this;
     }
 
+    /**
+     * Puts the key / values combination into the MutableBagIterableMultimap and returns
+     * the MutableBagIterableMultimap (this).
+     * @since 11.1
+     */
```
Comment (motlin) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java:42]: I think you only need javadoc at the top. You should check me on this, but I don't think that copy/pasting the Javadoc on a sub-interface changes the rendered html in any way.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java [hunk 3]
```diff
@@ -93,7 +95,29 @@ public boolean put(K key, V value)
...
+    }
+
+    protected void addAll(C existingValues, V[] values)
+    {
+        int currentSize = existingValues.size();
+        int newSize = ArrayIterate.addAllTo(values, existingValues).size();
+        this.addToTotalSize(newSize - currentSize);
+    }
+
+    protected C getIfAbsentPutCollection(K key)
```
Comment (motlin) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java:120]: Let's keep this method private, and use the casting approach in subclasses.

**Review by motlin — CHANGES_REQUESTED** [review]

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/MutableMultimap.java [hunk 4]
```diff
@@ -44,6 +44,22 @@ default MutableMultimap<K, V> withKeyValue(K key, V value)
...
+    {
+        if (values != null && values.length > 0)
+        {
+            for (V value : values)
+            {
+                this.put(key, value);
+            }
+        }
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (goldbal330) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/MutableMultimap.java]: Addressed.

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/list/MutableListMultimap.java [hunk 5]
```diff
@@ -37,6 +37,23 @@ default MutableListMultimap<K, V> withKeyValue(K key, V value)
         return this;
     }
 
+    /**
+     * Puts the key / values combination into the MutableListMultimap and returns
+     * the MutableListMultimap (this).
+     * @since 11.1
+     */
+    default MutableListMultimap<K, V> withKeyMultiValues(K key, V... values)
```
[the remark this answers is not in the document]
Comment (goldbal330) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/list/MutableListMultimap.java:41]: Addressed.

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java (hunk 1 above)
Comment (goldbal330) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java]: Addressed.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java (hunk 3 above)
Comment (goldbal330) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java]: Addressed.

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java (hunk 2 above)
Comment (goldbal330) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/multimap/bag/MutableBagIterableMultimap.java]: Addressed.

**Comment by motlin** [comment]
LGTM!

**Review by motlin — APPROVED** [review]

**Comment by donraab** [comment]
@goldbal330 Please rebase

**Pull request merged**